### ISSUE TITLE
validation dataset containing categorical value not in training dataset

### ISSUE THREAD

**Issue by dnwldnwl2** [jdb78/pytorch-forecasting#157]
- PyTorch-Forecasting version: latest
- PyTorch version: 1.7
- Python version: 3.7
- Operating System: debian

Hi. One of my column has unique values [c0, c1, c2, ..., c15, c16, h1, h2]. These values ​​are categorical values. I am trying to predict the rows denoted by h using the rows denoted by c. As h values ​​are not included in training, so an error occurs when creating a validation set. I currently have a dataset excluding this column, is there any good way to include this column values?

**Comment by jdb78** [comment]
You can try the following: Pass the argument
`categorical_encoders={name_of_the_column: NaNLabelEncoder(add_nan=True)}` to the TimeSeriesDataSet

Its documentation is here:
https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.data.encoders.NaNLabelEncoder.html#pytorch_forecasting.data.encoders.NaNLabelEncoder

**Comment by dnwldnwl2** [comment]
> 
> 
> You can try the following: Pass the argument
> `categorical_encoders={name_of_the_column: NaNLabelEncoder(add_nan=True)}` to the TimeSeriesDataSet
> 
> Its documentation is here:
> https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.data.encoders.NaNLabelEncoder.html#pytorch_forecasting.data.encoders.NaNLabelEncoder

It works. Thanks!
